web/modules/custom/joinup_group/src/Form/RdfDeleteForm.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_group\Form;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Link;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\rdf_entity\Form\RdfDeleteForm as OriginalForm;
+use Drupal\rdf_entity\RdfInterface;
+
+/**
+ * Prevents deletion of collections if there is an existing child solution.
+ */
+class RdfDeleteForm extends OriginalForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion(): TranslatableMarkup {
+    $entity = $this->getEntity();
+    if ($entity->bundle() !== 'collection' || $entity->get('field_ar_affiliates')->isEmpty()) {
+      return $this->t('Are you sure you want to delete entity %name?', [
```

Comment:
We can do more, we can be more specific as now we have the undefined singular label. The word "entity" is too vague for an end-user: `Are you sure you want to delete @type %name?`. Then `@type` is set to `$entity->get('rid')->entity->getSingularLabel()`.